### PR TITLE
AG-12522 - Fix firing of click event after drag.

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-12522

### CODE REVIEW

**Review by olegat — CHANGES_REQUESTED** [review]
> [!CAUTION]
> This solution is not compliant with WCAG 2 Success Criteria 2.2.3:
> > 2.2.3 No Timing, Level AAA
> > Timing is not an essential part of the event or activity presented by the content, except for non-interactive synchronized media and real-time events.
> https://www.w3.org/WAI/WCAG22/quickref/?versions=2.1#no-timing

As we discussed on Slack, another solution could be to record `mousemove` events in `InteractionManager` as we do with `mousedown` and `mouseup` events. AFAICT, this approach would be compliant with WCAG 2. Are there any side-effects from this approach that I am unaware of?

**Review by olegat — CHANGES_REQUESTED** [review]
[deleted]

**Comment by alantreadway** [comment]
> Would it be better to merge this breaking change `next` rather than `latest`?

We don't document the expected type of `event`, so although it is a change in behaviour, it is still correct from an API/contract perspective.

File: packages/ag-charts-community/src/chart/interaction/interactionManager.ts [hunk 1]
```diff
@@ -374,15 +373,15 @@ export class InteractionManager extends InteractionStateListener<InteractionType
                     return;
                 }
                 this.mouseDown = false;
-                return this.recordUp(event) ? 'drag-end' : undefined;
+                return this.recordUp(event) ? 'drag-end' : 'click';
```
Comment (olegat) [inline packages/ag-charts-community/src/chart/interaction/interactionManager.ts:376]: test

**Review by olegat — APPROVED** [review]
LGTM

**Pull request merged**